Polyclinic/Polyclinic.Api/Controllers/AnalyticsController.cs

```diff
@@ -0,0 +1,129 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Polyclinic.Contracts;
+using Polyclinic.Contracts.Appointments;
+using Polyclinic.Contracts.Doctors;
+using Polyclinic.Contracts.Patients;
+
+namespace Polyclinic.Api.Controllers;
+
+/// <summary>
+/// Controller for handling analytical data and reports
+/// </summary>
+public class AnalyticsController(IAnalyticsService service, ILogger<AnalyticsController> logger) : Controller
+{
+    /// <summary>
+    /// Retrieves doctors with experience at least specified number of years
+    /// </summary>
+    /// <param name="minYears">Minimum years of experience required</param>
+    /// <returns>List of doctors meeting the experience criteria</returns>
+    [HttpGet("doctors/experienced/{minYears}")]
+    [ProducesResponseType(200)]
+    [ProducesResponseType(204)]
+    [ProducesResponseType(500)]
+    public ActionResult<List<DoctorDto>> GetDoctorsWithExperienceAtLeast(int minYears)
+    {
+        try
+        {
+            var res = service.GetDoctorsWithExperienceAtLeast(minYears);
+            return res.Count > 0 ? Ok(res) : NoContent();
+        }
+        catch (Exception ex)
+        {
+            logger.LogError("Error in GetDoctorsWithExperienceAtLeast: {Message}", ex.Message);
+            return StatusCode(500, $"{ex.Message}\n\r{ex.InnerException?.Message}");
```

Comment:
Ошибки лучше логировать немного по другому. Первым аргументом лучше передавать всё исключение.
Также в ответе лучше не передавать все тонкости в InnerException (написал на русском, не хочу заморачиваться, сделаете по аналогии как вам больше нравится).
```suggestion
            logger.LogError(ex, "Error in GetDoctorsWithExperienceAtLeast: {Message}", ex.Message);
            return StatusCode(500, "Не удалось получить список врачей");
```